bert/README.md

```diff
@@ -0,0 +1,233 @@
+# This will construct a `StreamingC4` dataset from your `val` split,
+# pass it into a PyTorch Dataloader, and iterate over it and print samples.
+# Since remote and local are set to the same path, no streaming/copying takes place.
+python src/data_c4.py ./my-copy-c4 ./my-copy-c4
+
+# This will do the same thing, but stream data from {remote} -> {local}.
+# The remote path can be a filesystem or object store URI.
+python src/data_c4.py ./my-copy-c4 /tmp/cache-c4
+python src/data_c4.py s3://my-bucket/my-copy-c4 /tmp/cache-c4
+```
+
+# Training
+
+Now that you've installed dependencies and built a local copy of the C4 dataset, let's start training! We'll start with MLM pre-training on C4.
+
+**Please remember** to edit the `data_remote` and `data_local` paths in your YAML to point to your local C4 copy.
+Our streaming dataloader always streams from `data_remote` -> `data_local`, and if both paths are the same,
+then no extra copying is done.
+
+**Also remember** that if you only downloaded the `val` split, you need to make sure your train_dataloader is pointed at that split.
+Just change `split: train` to `split: val` in your YAML.
+This is already done in the testing YAML `yamls/test-cpu/main.py`, which you can also use to test your configuration.
+
+## MLM pre-training
+
+We run the `main.py` script using our `composer` launcher, which generates N processes (1 process per GPU device).
+
+If training on a single node, the `composer` launcher will autodetect the number of devices, so all you need to do to pre-train a Mosaic BERT is run:
+
+```bash
+composer main.py yamls/mosaic-bert-base-uncased.yaml
+```
+
+**Note:** The `yamls/*.yaml` files are intended to be used with `main.py`.
+
+**Please remember** to modify the reference YAMLs (e.g., `yamls/mosiac-bert-base-uncased.yaml`) to customize saving and loading locations -- only the YAMLs in `yamls/test-cpu/` are ready to use out-of-the-box.
+
+## GLUE fine-tuning
+
+The GLUE benchmark measures the average performance across 8 NLP classification tasks. This performance is typically used to evaluate the quality of the pre-training: once you have a set of weights from your MLM task, you fine-tune those weights separately for each task and average the results, with higher averages indicating higher pre-training quality.
+
+To handle this complicated fine-tuning pipeline, we provide the `glue.py` script, which supports parallelizing each of these fine-tuning jobs across all the GPUs on your node.
+
+Once you have modified the YAMLs in `yamls/glue/` to reference your pre-trained checkpoint as the GLUE starting point, run the `glue.py` script using the standard `python` launcher (we don't use the `composer` launcher here because `glue.py` does its own multi-process orchestration):
+
+```bash
+composer glue.py yamls/glue/mosaic-bert-base-uncased.yaml
+```
+
+Aggregate GLUE scores will be printed out at the end of the script and can also be tracked using Weights and Biases, if enabled via the YAML.
```

Comment:
suggest "using Weights and Biases" -> "using Weights and Biases (any of the other (composer supported loggers)[https://docs.mosaicml.com/en/latest/trainer/logging.html#available-loggers] can be added easily as well)"